api/src/services/authorization.js

```diff
@@ -217,26 +233,34 @@ const findContactsByReplicationKeys = (replicationKeys) => {
   }
 
   replicationKeys = _.uniq(replicationKeys);
-  const keys = replicationKeys.map(id => ['shortcode', id]);
+  const keys = [];
+  replicationKeys.forEach(id => keys.push(['shortcode', id], ['tombstone-shortcode', id]));
 
   return db.medic
     .query('medic-client/contacts_by_reference', { keys })
     .then(result => {
-      const docIds = replicationKeys.map(replicationKey => {
-        const found = result.rows.find(row => row.key[1] === replicationKey);
-        return found && found.id || replicationKey;
+      let docIds = [];
+      replicationKeys.forEach(replicationKey => {
+        const found = result.rows.filter(row => row.key[1] === replicationKey);
+        docIds.push(...found.length ? found.map(row => row.id) : [replicationKey]);
```

Comment:
I find this section confusing. I think it would help if the `map` was chained after the `filter` rather than buried inside a ternary. Also the spread operator on the ternary result is hard to parse. And there's no point creating an array for `replicationKey` then spreading it again.

I know it's more lines but what about something like...

```js
const keys = result.rows.filter(row => row.key[1] === replicationKey).map(row => row.id);
if (keys.length) {
  docIds.push(...keys);
} else {
  docIds.push(replicationKey);
}
```

Reply:
Yea, agree. I'll change it tomorrow. I originally wrote it similarly. I secretly love one-liners.